casbin/src/main.rs

```diff
@@ -0,0 +1,58 @@
+use casbin::{DefaultModel, Enforcer, FileAdapter, RbacApi};
+use std::boxed::Box;
+use std::io;
+use std::sync::RwLock;
+
+use actix_web::{middleware, web, App, HttpRequest, HttpResponse, HttpServer};
+
+/// simple handle
+async fn success(
+    enforcer: web::Data<RwLock<Enforcer>>,
+    req: HttpRequest,
+) -> HttpResponse {
+    let mut e = enforcer.write().unwrap();
+    println!("{:?}", req);
+    assert_eq!(vec!["data2_admin"], e.get_roles_for_user("alice", None));
+
+    HttpResponse::Ok().body(format!("Num of requests: {}", 1))
```

Comment:
What's the intention of this `format!` here?

Reply:
Obviously, just output something to indicate that this access request is valid.

Reply:
So, it's unnecessary to use `format!` and hard-coded value, it should be just a str.

Reply:
Got it.

I think another thing worth discussing is whether I should use `assert_eq!`. If it fails, it may cause the program to not work properly.

Reply:
> I think another thing worth discussing is whether I should use assert_eq!. If it fails, it may cause the program to not work properly.

I'm fine with that since it's just an example and the purpose is obvious. But feel free to extract them.